spec/vmpooler/api/v1_spec.rb

```diff
@@ -432,6 +432,34 @@ def app()
         end
       end
     end
+
+    describe 'POST /vm/:hostname/snapshot' do
+        it 'creates a snapshot' do
+          expect(redis).to receive(:sadd)
+
+          post "#{prefix}/vm/testhost/snapshot"
+
+          expect(last_response.header['Content-Type']).to eq('application/json')
+          expect(last_response.body).to include('"ok": true')
+          expect(last_response.body).to include('"testhost": {')
+          expect(last_response.body).to include('"snapshot": "')
```

Comment:
The above 3 lines can be altered to check for key existence if you parse the body as JSON:

```
json_response = JSON.parse(last_response.body)
expect(json_response['ok']).not_to be_nil
```